deepspeed/runtime/config.py

```diff
@@ -521,16 +523,14 @@ def write_config(self, filename):
 
 
 class DeepSpeedConfig(object):
-    def __init__(self, json_file, mpu=None, param_dict=None):
+    def __init__(self, ds_config: Union[str, dict], mpu=None):
         super(DeepSpeedConfig, self).__init__()
-
-        if param_dict is None:
+        self._param_dict = ds_config
+        if not isinstance(ds_config, dict) and os.path.exists(ds_config):

```

Comment:
this would break if the path doesn't exist, since json won't be loaded and we are now missing the dict completely.

so this needs `else: raise ValueError(...)` if the check fails.

Reply:
Let me know if it look good now!

Reply:
It's perfect, thank you for adding the check, @SeanNaren